File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.java
class Solution {
    class Distances{
        int[] coordinates;
        double distanceFromOrigin;

        Distances(int[] coordinates, double distanceFromOrigin){
            this.coordinates = coordinates;
            this.distanceFromOrigin = distanceFromOrigin;
        }
    }
    double helper(int i, int j){
        double temp = Math.pow(i, 2) + Math.pow(j, 2);
        return Math.sqrt(temp);
    }
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<Distances> qu = new PriorityQueue<>(new Comparator<Distances>() {
            public int compare(Distances d1, Distances d2) {
                return Double.compare(d1.distanceFromOrigin, d2.distanceFromOrigin);
            }
        });
        int i = 0;
        int j = 0;
        int[][] res = new int[k][2];
        while(i < points.length){
            double currDistance = helper(points[i][0], points[i][1]);
            Distances curr = new Distances(points[i], currDistance);
            qu.offer(curr);
            if(qu.size() > points.length - k){
                res[j] = qu.poll().coordinates;
                j++;
            }
            i++;
        }
        return res;
}}

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        def helper(point):
            xcoor = point[0]
            ycoor = point[1]
            return math.sqrt(math.pow(xcoor,2) + math.pow(ycoor,2))
        heap = []
        for i in range(len(points)):
            heapq.heappush(heap,(helper(points[i]), i))
        result = []
        for i in range(k):
            result.append(points[heapq.heappop(heap)[1]])
        return result

        
        